packages/core-sdk/src/resources/dispute.ts

```diff
@@ -251,23 +278,194 @@ export class DisputeClient {
+                    internalType: "struct OptimisticOracleV3Interface.EscalationManagerSettings",
+                    name: "escalationManagerSettings",
+                    type: "tuple",
+                  },
+                  {
+                    internalType: "address",
+                    name: "asserter",
+                    type: "address",
+                  },
+                  {
+                    internalType: "uint64",
+                    name: "assertionTime",
+                    type: "uint64",
+                  },
+                  {
+                    internalType: "bool",
+                    name: "settled",
+                    type: "bool",
+                  },
+                  {
+                    internalType: "contract IERC20",
+                    name: "currency",
+                    type: "address",
+                  },
+                  {
+                    internalType: "uint64",
+                    name: "expirationTime",
+                    type: "uint64",
+                  },
+                  {
+                    internalType: "bool",
+                    name: "settlementResolution",
+                    type: "bool",
+                  },
+                  {
+                    internalType: "bytes32",
+                    name: "domainId",
+                    type: "bytes32",
+                  },
+                  {
+                    internalType: "bytes32",
+                    name: "identifier",
+                    type: "bytes32",
+                  },
+                  {
+                    internalType: "uint256",
+                    name: "bond",
+                    type: "uint256",
+                  },
+                  {
+                    internalType: "address",
+                    name: "callbackRecipient",
+                    type: "address",
+                  },
+                  {
+                    internalType: "address",
+                    name: "disputer",
+                    type: "address",
+                  },
+                ],
+                internalType: "struct OptimisticOracleV3Interface.Assertion",
+                name: "",
+                type: "tuple",
+              },
+            ],
+            stateMutability: "view",
+            type: "function",
+          },
+        ],
+        functionName: "getAssertion",
+        args: [request.assertionId],
+      });
       const counterEvidenceHash = convertCIDtoHashIPFS(request.counterEvidenceCID);
       const encodedData = this.arbitrationPolicyUmaClient.disputeAssertionEncode({
         assertionId: request.assertionId,
         counterEvidenceHash,
       });
-      const txHash = await ipAccount.execute({
-        to: encodedData.to,
-        value: 0n,
-        data: encodedData.data,
-        operation: 0,
+      // Approve ipAccount to transfer wrappedIp, cannot use ipAccount.executeBatch because msg.sender is ipAccount with same address as spender.
```

Comment:
> cannot use ipAccount.executeBatch because msg.sender is ipAccount with same address as spender.

i think this part needs a bit more clarity. I understand we have to do this separate because we need the client wallet to approve the ipAccount to use the WIP from the client wallet to pay for the bond right? We should include that comment here.